python/dgl/graphbolt/impl/fused_csc_sampling_graph.py

```diff
@@ -974,9 +985,31 @@ def _pin(x):
 
     def pin_memory_(self):
         """Copy `FusedCSCSamplingGraph` to the pinned memory in-place."""
+        # torch.Tensor.pin_memory() is not an inplace operation. To make it
+        # truly in-place, we need to use cudaHostRegister. Then, we need to use
+        # cudaHostUnregister to unpin the tensor in the destructor.
+        # https://github.com/pytorch/pytorch/issues/32167#issuecomment-753551842
+        cudart = torch.cuda.cudart()
 
         def _pin(x):
-            return x.pin_memory() if hasattr(x, "pin_memory") else x
+            if hasattr(x, "pin_memory_"):
+                x.pin_memory_()
+            elif (
+                isinstance(x, torch.Tensor)
+                and not x.is_pinned()
+                and x.device.type == "cpu"
+                and x.is_contiguous()
```

Comment:
If a tensor is not contiguous, it will cause some unexpected errors if users try to pin it. So I'd suggest to raise an error if the device type is not GPU or the tensor is not contiguous.

Reply:
Addressed it. Thanks!